backend/canvas_app_explorer/canvas_lti_manager.py

```diff
@@ -0,0 +1,21 @@
+from typing import List
+from canvasapi import Canvas
+from .data_class import ExternalTool
+
+class CanvasLtiManager:
+    def __init__(self, API_URL: str, API_KEY: str, course_id: int):
+        self.course_id = course_id
+        self.requestor = Canvas(API_URL, API_KEY)
+
+    def get_tools_available_in_course(self) -> List[ExternalTool]:
+        ex_tools = []
+        course = self.requestor.get_course(self.course_id)
+        for tab in course.get_tabs():
+            if 'external_tools' in tab.html_url: #filter external tools from all tabs
+                ex_tools.append(ExternalTool(
+                    label=tab.label,
+                    id=int(tab.id.split('_')[-1]),
```

Comment:
Yeah, I agree and thought about this too. There's either going to be an full exception here (if they happen to change this pattern) or we'd have to do a try/catch.

This is [used in many places in Canvas ](https://github.com/instructure/canvas-lms/search?q=context_external_tool_) so seems like a big change that we'd probably be aware of if they ever did try to change it. 

Though even if they did something like removed the underscores and just made the id a int this would still work. The only way this would fail is if they changed the pattern completely which just looks unlikely.